examples/server/server.cpp

```diff
@@ -2412,6 +2435,66 @@ inline void signal_handler(int signal) {
     shutdown_handler(signal);
 }
 
+static void handle_tasks(
+    bool stream,
+    httplib::Response & res,
+    server_context & ctx_server,
+    const std::function<std::unordered_set<int>(const std::function<bool()> &)> & create_tasks,
+    const std::function<void(const std::unordered_set<int> &, httplib::DataSink & sink)> & payload)
+{
+    struct State {
+        std::unordered_set<int> task_ids;
+        bool is_sink_valid = true;
+    };
+    auto state = std::make_shared<State>();
```

Comment:
To be honest, I would prevent using shared_ptr whenever possible. If you search google for "why shared_ptr is bad", you will know.